readthedocs/doc_builder/backends/mkdocs.py

```diff
@@ -198,6 +199,8 @@ def build(self):
             '--site-dir', self.build_dir,
             '--config-file', self.yaml_file,
         ]
+        if self.config.mkdocs.fail_on_warning:
+            build_command.append('--strict')
```

Comment:
Personally, I'd let users configure this in their [`mkdocs.yml`](https://www.mkdocs.org/user-guide/configuration/#strict) rather than adding an extra Read the Docs option.

Reply:
I'm +1 for this, I didn't know about that, Sphinx doesn't have something like that.

Reply:
I think you could define `setup()` function in `conf.py` and override `config.warningiserror = True` and I think it should work.